src/averageAges.js

```diff
@@ -37,7 +43,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWomen = !withChildren
+    ? people.filter(person => person.sex === 'f')
+    : people.filter(person =>
+      (people.find(child => child.mother === person.name)));
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

Method `some` would be better use for this purpose instead of `find`, because this way as I understand you need only boolean, not entire person object.